lib/compat.php

```diff
@@ -8,6 +8,104 @@
  * @package gutenberg
  */
 
+if ( ! function_exists( 'get_default_block_editor_settings' ) ) {
+	/**
+	 * Gets the default block editor settings.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_Post $post Global post object.
+	 *
+	 * @return array The default block editor settings.
+	 */
+	function get_default_block_editor_settings( $post = null ) {
+		/**
+		 * Filters the allowed block types for the editor, defaulting to true (all
+		 * block types supported).
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param bool|array $allowed_block_types Array of block type slugs, or
+		 *                                        boolean to enable/disable all.
+		 * @param WP_Post    $post                The post resource data.
+		 */
+		$allowed_block_types = apply_filters( 'allowed_block_types', true, $post );
```

Comment:
Thinking a bit more about it, the biggest concern I have is that existing filters assume that `$post` contains an object, like here:
https://developer.wordpress.org/reference/hooks/allowed_block_types/#comment-3176
We even promote it ourselves in the documentation:
https://developer.wordpress.org/block-editor/reference-guides/filters/block-filters/#hiding-blocks-from-the-inserter

We need to run this filter conditionally and provide an alternative for other screens.

Reply:
We can keep this filter in just "post editor", deprecate it and create a new generic one?

Reply:
This is what I ended up doing 👍🏻